test/unit/css.js

```diff
@@ -203,6 +203,47 @@ test( "css() explicit and relative values", 29, function() {
 	equal( $elem.css("opacity"), "1", "'+=0.5' on opacity (params)" );
 });
 
+test( "css() non-px relative values (gh-1711)", 10, function() {
+	var $parent = jQuery("#nothiddendiv"),
+		$child = jQuery("#nothiddendivchild", $parent),
+		widthCurr, units,
+		add = function( val, unit ) {
+			var str = ( val < 0 ? "-=" : "+=" ) + Math.abs(val) + unit;
+			$child.css( "width", str );
+			equal(
+				$child.width(),
+				Math.round( widthCurr += val * units[ unit ] ),
+				"'" + str + "' on width"
+			);
+		};
+
+	$parent.css({ height: 1, padding: 0, width: 400 });
+	$child.css({ height: 1, padding: 0 });
+
+	units = {
+		"px": 1,
+		"em": parseFloat( $child.css( "fontSize", "16px" ).css( "fontSize" )),
+		"pt": $child.css( "width", "100pt" ).width() / 100,
+		"pc": $child.css( "width", "100pc" ).width() / 100,
+		"cm": $child.css( "width", "100cm" ).width() / 100,
+		"mm": $child.css( "width", "100mm" ).width() / 100,
+		"%" : $child.css( "width", "50%" ).width() / 50
+	};
+
+	widthCurr = $child.width();
+
+	add( 25, "%");
+	add(-50, "%");
+	add( 10, "em");
+	add( 10, "pt");
+	add(-2.3, "pt");
+	add( 5, "pc");
+	add(-5, "em");
+	add(+2, "cm");
+	add(-15, "mm");
+	add(21, "px");
+});
+
```

Comment:
Could you also add some tests against `line-height` instead of `width`? That property has caused problems in the past, and I'd like to know it's solid.